eclipse/che
# Gh18838 factory devfile2

### What does this PR do?
Adds support for devfile v2 to /factory/resolve. Validation and property overrides are not working at this point.

### Screenshot/screencast of this PR

```
$ curl -k 'https://che-che.192.168.39.5.nip.io/api/factory/resolver' -H 'Content-Type: application/json' --data-raw '{"url":"https://github.com/sparkoo/spring-petclinic"}' | jq

  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   534  100   481  100    53   8745    963 --:--:-- --:--:-- --:--:--  9709
{
  "v": "4.0",
  "source": "devfile.yaml",
  "devfile": {
    "schemaVersion": "2.0.0",
    "metadata": {
      "name": "spring petclinic demo"
    },
    "projects": [
      {
        "name": "spring-petclinic",
        "git": {
          "remotes": {
            "origin": "https://github.com/sparkoo/spring-petclinic"
          },
          "checkoutFrom": {
            "revision": "master",
            "remote": "origin"
          }
        }
      }
    ],
    "components": [
      {
        "name": "mysql",
        "container": {
          "image": "quay.io/eclipse/che--centos--mysql-57-centos7:latest-e08ee4d43b7356607685b69bde6335e27cf20c020f345b6c6c59400183882764",
          "memoryLimit": "128M"
        }
      }
    ]
  }
}

```

### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/18838

### How to test this PR?
Install Che and call `/factory/resolve` che-server API with devfile v2. You can use my test repo https://github.com/sparkoo/spring-petclinic or any other you like with devfile v2.
Body of the request is `{"url": "<repo/devfile-url>"}` (see curl example above)


### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
